src/rewriter.fs

```diff
@@ -964,7 +969,29 @@ let rec private iterateSimplifyAndInline (options: Options.Options) optimization
             iterateSimplifyAndInline options optimizationPass (passCount + 1) li
         else li
 
-let simplify options li =
+let simplify (options: Options.Options) li =
+    let mutable forceInlineNextFunction = None
+    let processPragmas tl =
+        match tl with
+        | TLDirective ["#pragma function inline"] ->
+            forceInlineNextFunction <- Some true
+            None
+        | TLDirective ["#pragma function noinline"] ->
+            forceInlineNextFunction <- Some false
+            None
+        | Function (ft, _) as tl ->
+            match forceInlineNextFunction with
+            | Some true ->
+                options.trace $"{ft.fName.Loc}: pragma forces inlining of '{Printer.debugFunc ft}'"
+                ft.fName.ToBeInlined <- true
+            | Some false ->
+                options.trace $"{ft.fName.Loc}: pragma prevents inlining of '{Printer.debugFunc ft}'"
+                ft.fName.DoNotInline <- true
+            | None -> ()
+            forceInlineNextFunction <- None
+            Some tl
+        | tl -> Some tl
```

Comment:
What if `#pragma function inline` is not followed by a function?
Do we want to handle that case?

Reply:
We could emit a warning for unused function inline pragmas.